engine/apps/metrics_exporter/metrics_collectors.py

```diff
@@ -196,5 +203,28 @@ def get_buckets_with_sum(self, values):
             sum_value += value
         return buckets_values, sum_value
 
+    def _get_schedule(self, org_ids):
+        # Initialize the metric object with the updated metric name
+        metrics_schedule = GaugeMetricFamily(
+            METRICS_SCHEDULE, "Description of metric", labels=["schedule", "team", "username"]
+        )
+
+        # Iterate over organization IDs
+        for org_id in org_ids:
```

Comment:
I think this loop may be too expensive to run for multiple orgs/schedules (and make the scrape timeout). Have you considered using a similar approach to the other metrics keeping data in cache and just fetching from there when running the collector? Note that we already have a periodical task checking who is on-call for each schedule (`refresh_ical_file`) which already updates cache (`update_cached_oncall_users_for_schedule`). I think it should be possible to plug this there?

Reply:
Updated. I don’t know how to optimize it even more.